### PR TITLE
Changed name of entry type in layout files

### PR DESCRIPTION
**Pull request by oscargus**
For consistency with the recent versions of the code. Still backwards compatible though.

- [x] Change in CHANGELOG.md described

### CODE REVIEW

**Comment by oscargus** [comment]
I didn't see any information about this at the help pages. Maybe it is worthwhile pointing out the field name to use?

**Comment by Siedlerchr** [comment]
Not sure, but the sample code on writing CustomImporters should be then corrected, too?
https://help.jabref.org/en/CustomImports

**Comment by oscargus** [comment]
Not because of this PR, but clearly because of the importer refactoring.
Hadn't thought about that...

It has been `entrytype` for quite some time internally, so may make sense
to use the same. However, it struck me that there could be other formats
using `bibtextype` such as BibXML or whatever it is called. Not that it is
a problem but maybe it is good to be consistent...

**Comment by stefan-kolb** [comment]
An automatic migration for this would be good?!

**Comment by oscargus** [comment]
I'm not sure if we want to change peoples layout files... This is sort of similar as when FormatChars was replaced. At the moment there is a still support for the old name.

It would be nice to make people aware of it one way or the other though...

**Comment by oscargus** [comment]
I should change the default previews as well.

File: src/main/java/net/sf/jabref/gui/preftabs/FileTab.java [hunk 1]
```diff
@@ -285,10 +285,6 @@ public void storeSettings() {
             prefs.put(JabRefPreferences.NON_WRAPPABLE_FIELDS, nonWrappableFields.getText());
             updateSpecialFields = true;
         }
-        // If either of the two last entries were changed, run the update for special field handling:
```
Comment (koppor) [inline src/main/java/net/sf/jabref/gui/preftabs/FileTab.java]: Why was this code removed? OK, I don't even get the deleted comment, but maybe the code was useful.

Comment (oscargus) [inline src/main/java/net/sf/jabref/gui/preftabs/FileTab.java]: See `updateSpecialFieldHandling()` :-) 

It updates `private final Set<String> nonWrappableFields = new HashSet<>(5)` which is never used somewhere else...

**Comment by oscargus** [comment]
I updated the code to print a warning if `bibtextype` is used in a layout, with an encouragement to change it.

File: src/main/java/net/sf/jabref/logic/xmp/XMPUtil.java [hunk 2]
```diff
@@ -832,7 +832,7 @@ private static void writeToDCSchema(XMPSchemaDublinCore dcSchema,
              *
              * Description: Relationships to other documents.
              *
-             * Bibtex-Fields used: bibtextype
+             * Bibtex-Fields used: entrytype
              */
```
Comment (koppor) [inline src/main/java/net/sf/jabref/logic/xmp/XMPUtil.java]: Can the comments be merged?

**Comment by koppor** [comment]
Please adress my minor comment and merge.

I would have liked to see test cases for the layouts, but I think, this is too complicated for now.

Minor thing: We should render the article type in a similar way as in the `bib` file: `InProceedings` vs. `Inproceedings`:

grabbed_20160815-091644

grabbed_20160815-091659

File: src/main/java/net/sf/jabref/logic/xmp/XMPUtil.java (hunk 2 above)
Comment (oscargus) [inline src/main/java/net/sf/jabref/logic/xmp/XMPUtil.java]: Probably. But to be honest I didn't understand the code more than I'm quite sure that I should change the comment.

**Comment by oscargus** [comment]
The rendering in line 113 of MainTableColumn, but I have no idea where the camel cased entry types reside (should be easy to find though).

**Comment by oscargus** [comment]
I fixed rendering in the main table and added a formatter, `EntryTypeFormatter` for camel casing entry type names.

File: src/main/java/net/sf/jabref/logic/xmp/XMPUtil.java (hunk 2 above)
Comment (oscargus) [inline src/main/java/net/sf/jabref/logic/xmp/XMPUtil.java]: That comment was quite misplaced. I correct the actual type comment.

File: src/main/java/net/sf/jabref/logic/layout/format/EntryTypeFormatter.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+public class EntryTypeFormatter implements LayoutFormatter {
+
+    @Override
+    public String format(String fieldText) {
+        // Fancy typesetting
+        BibEntry entry = new BibEntry();
+        entry.setType(fieldText);
+        TypedBibEntry typedEntry = new TypedBibEntry(entry, BibDatabaseMode.BIBLATEX);
+        return typedEntry.getTypeForDisplay();
+
```
Comment (koppor) [inline src/main/java/net/sf/jabref/logic/layout/format/EntryTypeFormatter.java]: no empty line :innocent:

File: src/main/java/net/sf/jabref/logic/layout/LayoutEntry.java [hunk 4]
```diff
@@ -235,9 +236,15 @@ public String doLayout(BibEntry bibtex, BibDatabase database, Optional<Pattern>
...
-        if ("bibtextype".equals(text)) {
+        if (BibEntry.TYPE_HEADER.equals(text)) {
+            fieldEntry = bibtex.getType();
+        } else if (BibEntry.OBSOLETE_TYPE_HEADER.equals(text)) {
+            LOGGER.warn("'" + BibEntry.OBSOLETE_TYPE_HEADER
+                    + "' is an obsolete name for the entry type. Please update your layout to use '"
+                    + BibEntry.TYPE_HEADER + "' instead.");
             fieldEntry = bibtex.getType();
         } else {
+
```
Comment (koppor) [inline src/main/java/net/sf/jabref/logic/layout/LayoutEntry.java]: Why the empty line?

File: src/main/java/net/sf/jabref/logic/layout/format/EntryTypeFormatter.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+import net.sf.jabref.logic.layout.LayoutFormatter;
+import net.sf.jabref.model.database.BibDatabaseMode;
+import net.sf.jabref.model.entry.BibEntry;
+
+
+public class EntryTypeFormatter implements LayoutFormatter {
+
+    @Override
+    public String format(String fieldText) {
+        // Fancy typesetting
```
Comment (koppor) [inline src/main/java/net/sf/jabref/logic/layout/format/EntryTypeFormatter.java]: Could you remove that line and add a class comment with an example (above line 9)?

File: src/main/java/net/sf/jabref/logic/layout/format/EntryTypeFormatter.java [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+import net.sf.jabref.logic.TypedBibEntry;
+import net.sf.jabref.logic.layout.LayoutFormatter;
+import net.sf.jabref.model.database.BibDatabaseMode;
+import net.sf.jabref.model.entry.BibEntry;
+
+
+public class EntryTypeFormatter implements LayoutFormatter {
+
+    @Override
+    public String format(String fieldText) {
```
Comment (koppor) [inline src/main/java/net/sf/jabref/logic/layout/format/EntryTypeFormatter.java]: Could you state that here the entryType as string is expected? Maybe rename the parameter form `fieldText` to `entryType`?

File: src/test/java/net/sf/jabref/logic/layout/format/EntryTypeFormatterTest.java [hunk 7]
```diff
@@ -0,0 +1,33 @@
+package net.sf.jabref.logic.layout.format;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import static org.junit.Assert.assertEquals;
+
+public class EntryTypeFormatterTest {
```
Comment (koppor) [inline src/test/java/net/sf/jabref/logic/layout/format/EntryTypeFormatterTest.java]: :+1:

File: src/main/java/net/sf/jabref/logic/layout/format/EntryTypeFormatter.java (hunk 6 above)
Comment (oscargus) [inline src/main/java/net/sf/jabref/logic/layout/format/EntryTypeFormatter.java]: Ahh, missed your comment. Will do that in next PR.

File: src/main/java/net/sf/jabref/logic/layout/format/EntryTypeFormatter.java (hunk 5 above)
Comment (oscargus) [inline src/main/java/net/sf/jabref/logic/layout/format/EntryTypeFormatter.java]: Same here.

File: src/main/java/net/sf/jabref/logic/layout/LayoutEntry.java (hunk 4 above)
Comment (oscargus) [inline src/main/java/net/sf/jabref/logic/layout/LayoutEntry.java]: Hmm, no idea...

**Pull request merged**